api/v1alpha4/types.go

```diff
@@ -724,3 +724,13 @@ type SpotMarketOptions struct {
 	// +kubebuilder:validation:pattern="^[0-9]+(\.[0-9]+)?$"
 	MaxPrice *string `json:"maxPrice,omitempty"`
 }
+
+// MachineAMIType specifies which AWS AMI to use for a AWSMachine and AWSMachinePool.
+type MachineAMIType string
+
+const (
+	// Al2x86_64 is the default AMI type.
+	Al2x86_64 MachineAMIType = "AL2_x86_64"
```

Comment:
I wonder if we should have names that are a bit more user-friendly. Perhaps we could have `AmazonLinux` and `AmazonLinuxGPU` or something similar.  It would be better UX, and also we have an open ticket to support Bottlerocket as well.

Reply:
done